### PR TITLE
Corner cases catches

### PR DESCRIPTION
**Pull request by jd-lara**
This PR addresses some bugs to catch corner cases exposed by the NTP data set.

### CODE REVIEW

File: src/system_utils.jl [hunk 1]
```diff
@@ -11,15 +11,16 @@ function _find_subnetworks(sys::PSY.System)
...
 
 """
 Finds the subnetworks in a system using Depth First Search (DFS). Returns a dictionary keyed
 by the reference bus of the subnetworks if they exist
 """
 function find_subnetworks(sys::PSY.System)
-    sbn, ref_bus_positions = _find_subnetworks(sys)
-    return assign_reference_buses(sbn, ref_bus_positions)
+    sbn, ref_buses = _find_subnetworks(sys)
+    return assign_reference_buses!(sbn, ref_buses)
```
Comment (GabrielKS) [inline src/system_utils.jl:25]: All the other existing `assign_reference_buses` calls have been edited to add a `bus_ax_ref` — this one doesn't need one?

File: src/common.jl [hunk 2]
```diff
@@ -336,6 +333,15 @@ function assign_reference_buses(
     return bus_groups
 end
 
+function assign_reference_buses!(
+    subnetworks::Dict{Int, Set{Int}},
+    ref_bus_positions::Set{Int},
+    bus_lookup::Dict{Int, Int},
+)
+    ref_buses = [k for (k, v) in bus_lookup if v in ref_bus_positions]
+    assign_reference_buses!(subnetworks, ref_buses)
```
Comment (GabrielKS) [inline src/common.jl:342]:
```suggestion
    return assign_reference_buses!(subnetworks, ref_buses)
```

A [style thing](https://github.com/invenia/BlueStyle#method-definitions) to make it clear we care about the return value

File: src/network_radial_reduction.jl [hunk 3]
```diff
@@ -99,6 +99,13 @@ function _new_parent(
             reverse_bus_map,
         )
         new_parent_bus_number = reverse_bus_map[new_parent_val]
+        # This check is meant to capture cases of a full radial network which can happen in
+        # system with small islands that represent larger interconnected areas.
+        if length(SparseArrays.nzrange(A, new_parent_val)) < 2
+            @warn "Bus $parent_bus_number Parent $new_parent_bus_number is a leaf node. Indicating there is an island."
+            push!(bus_reduction_map_index[parent_bus_number], new_parent_bus_number)
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply src/network_radial_reduction.jl:106]: or a `# TODO` or issue if this is too complicated to do on the current timeframe

File: src/common.jl [hunk 4]
```diff
@@ -336,6 +333,15 @@ function assign_reference_buses(
     return bus_groups
 end
 
+function assign_reference_buses!(
```
Comment (GabrielKS) [inline src/common.jl:336]: Ideally we'd get a test case that demonstrates what breaks without this new layer of indirection — if too complicated to do now but potentially feasible in the future, add a `# TODO` or GitHub Issue

**Review by GabrielKS — CHANGES_REQUESTED** [review]
A few typos and minor things, some requests either for tests or notes about the lack of tests.

File: src/network_radial_reduction.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (GabrielKS) [reply src/network_radial_reduction.jl:106]: Codecov [concurs](https://app.codecov.io/gh/NREL-Sienna/PowerNetworkMatrices.jl/pull/70?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=NREL-Sienna) and points out a few other untested regions

File: src/common.jl [hunk 5]
```diff
@@ -307,25 +307,22 @@ end
 """
 !!! MISSING DOCUMENTATION !!!
 """
-function assign_reference_buses(
+function assign_reference_buses!(
     subnetworks::Dict{Int, Set{Int}},
-    ref_bus_positions::Set{Int},
+    ref_buses::Vector{Int},
```
[the remark this answers is not in the document]
Comment (jd-lara) [reply src/common.jl:312]: because before it was incorrectly using the ref buses positions and not the ref buses values.

File: src/network_radial_reduction.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (jd-lara) [reply src/network_radial_reduction.jl:106]: I will create an issue to implement the test. This is a corner case catch.

File: src/network_radial_reduction.jl [hunk 6]
```diff
@@ -120,7 +127,11 @@ function _reverse_search(
     reverse_line_map::Dict{Int64, String},
     radial_branches::Set{String},
     reverse_bus_map::Dict{Int, Int},
+    ref_bus_positions::Set{Int},
 )
+    if j ∈ ref_bus_positions
```
[the remark this answers is not in the document]
Comment (jd-lara) [reply src/network_radial_reduction.jl:132]: We never want to reduce the reference buses. It creates issues depending on the model once we remove tha bus that is meant to absorb all the power differences.

File: src/system_utils.jl (hunk 1 above)
Comment (jd-lara) [reply src/system_utils.jl:25]: in some cases we already have the references buses so there is no need to recalculate them based on the ref_ax and location.

File: src/common.jl (hunk 2 above)
Comment (jd-lara) [reply src/common.jl:342]: sorry, I forgot this one

File: src/network_radial_reduction.jl [hunk 7]
```diff
@@ -120,7 +127,11 @@ function _reverse_search(
...
+        return
+    end
     j_bus_number = reverse_bus_map[j]
     pop!(bus_reduction_map_index, j_bus_number)
     reducion_set = Set{Int}(j_bus_number)
```
[the remark this answers is not in the document]
Comment (GabrielKS) [reply src/network_radial_reduction.jl]: Need to also rename on line 143 and line 147. GitHub won't let me make the suggestion on 143

**Comment by GabrielKS** [comment]
Added https://github.com/NREL-Sienna/PowerNetworkMatrices.jl/issues/71 for the tests
